src/cmd/api/controllers/video_archive_handler.go

```diff
@@ -0,0 +1,102 @@
+package controllers
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/Sogilis/Voogle/src/pkg/uuidgenerator"
+
+	"github.com/Sogilis/Voogle/src/cmd/api/db/dao"
+	"github.com/Sogilis/Voogle/src/cmd/api/models"
+)
+
+type VideoArchiveVideoHandler struct {
+	VideosDAO *dao.VideosDAO
+	UUIDGen   uuidgenerator.IUUIDGenerator
+}
+
+// VideoArchiveVideoHandler godoc
+// @Summary Archive video
+// @Description Archive video
+// @Tags status
+// @Accept plain
+// @Produce plain
+// @Param id path string true "Video ID"
+// @Success 200 {string} string "OK"
+// @Failure 400 {object} object
+// @Failure 404 {object} object
+// @Failure 500 {object} object
+
+// @Router /api/v1/videos/{id}/archive [put]
+func (v VideoArchiveVideoHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	log.Debug("PUT VideoArchiveVideoHandler - parameters ", vars)
+
+	id, exist := vars["id"]
+	if !exist {
+		log.Error("Missing video id")
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	if !v.UUIDGen.IsValidUUID(id) {
+		log.Error("Invalid id")
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	video, err := v.VideosDAO.GetVideo(r.Context(), id)
+	if err != nil {
+		log.Error("Cannot found video : ", err)
+		if errors.Is(err, sql.ErrNoRows) {
+			w.WriteHeader(http.StatusNotFound)
+		} else {
+			w.WriteHeader(http.StatusInternalServerError)
+		}
+		return
+	}
+
+	statusCode, err := v.archiveVideo(r.Context(), video)
+	if err != nil {
+		w.WriteHeader(statusCode)
+		return
+	}
+}
+
+func (v VideoArchiveVideoHandler) archiveVideo(ctx context.Context, video *models.Video) (int, error) {
+	tx, err := v.VideosDAO.DB.BeginTx(ctx, nil)
+	if err != nil {
+		log.Error("Cannot open new database transaction : ", err)
+		return http.StatusInternalServerError, err
+	}
+
+	video.Status = models.ARCHIVE
+
```

Comment:
Do I need to modify `video.UpdatedAt` ? And if I do, how to do it ?

Reply:
Yes. We can do it in api/db/dao, in the UpdateVideo function. Each update should update the video.UpdatedAt field

Reply:
I opened `api/db/dao/video.go` but I can't see where the `Video.UpdatedAt` field is updated to the new value.

Plus, there is a check a the end for assuring there was only 1 row affected so that's a little bit confusing.

Reply:
in file `src/cmd/api/db/dao/video.go` : `updated_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,`. It updates itself.